manage

```diff
@@ -126,19 +126,55 @@ babel.sha256sum() {
     grep "msgid" "searx/translations/messages.pot" | sort | sha256sum | cut -f1 -d ' '
 }
 
+weblate.merge.push.updates() {
+    (   set -e
+        pyenv.cmd wlc lock
+        if ! git remote get-url weblate; then
+            git remote add weblate https://weblate.bubu1.eu/git/searxng/searxng/
+        fi
+        git remote update weblate
+        git merge weblate/master
+        babel.compile
+        git add searx/translations
+        git commit -m "[translations] update messages.mo files"
+        git push
+    )
+    exitcode=$?
+    (   # make sure to always unlock weblate
+        set -e
+        pyenv.cmd wlc unlock
+    )
+    dump_return $exitcode
+}
+
 ci.babel.update() {
     local sha_before
     (   set -e
         sha_before="$(babel.sha256sum)"
         babel.extract
         if [ "$(babel.sha256sum)" = "${sha_before}" ]; then
             build_msg BABEL 'no changes detected, exiting'
-            return 1
+            return 0
         fi
+
+        # call weblate.merge.push.updates right away
+        git stash push "searx/translations/messages.pot"
+        weblate.merge.push.updates babel.update
+
+        # then update messages.pot while weblate is locked
+        pyenv.cmd wlc lock
+        git stash pop
         babel.update
-        build_msg BABEL 'update done, edit .po files if required and run babel.compile'
+        git add searx/translations
+        git commit -m "[translations] update messages.pot and messages.po files"
+        git push
     )
-    dump_return $?
+    exitcode=$?
+    (   # make sure to always unlock weblate
+        set -e
+        pyenv.cmd wlc unlock
+    )
+    dump_return $exitcode
```

Comment:
I am not familiar with the weblate workflows. Can add you comments to the `ci.babel.update()` and `weblate.merge.push.update()`  .. it might be good to describe what these function do in a 1., 2., 3. ... order.

Reply:
Sum up:
* weblate keeps a clone of the master branch of searxng/searxng.
* it reads searx/translations/messages.pot to get the text to translate
* it reads searx/translations/*/LC_MESSAGES/messages.po to get the existing translations
* it updates & commit searx/translations/*/LC_MESSAGES/messages.po  to persist new translations

So if the CI here changes messages.pot or messages.po, weblate must be locked = no change = no new commit on the weblate clone.

If messages.pot can't be updated from the code directly: weblate will have its own version and it will requires a conflict resolution.

So if messages.pot has some changes, the workflow is 
* lock weblate
* stash to remove messages.pot from the CI clone.
* pull weblate changes (weblate.merge.push.updates function)
  * merge the commits from weblate (only the .po files)
  * build the .mo files, commit
  * push the commits to the master branch
* stash pop to get back messages.pot 
* update the message.po from messages.pot (actually I'm not sure this is necessary: perhaps weblate take care of that)
* commit (messages.pot & message.po files)
* push the commit
* weblate unlock (like a try.... finally : it must be call to allow changes on weblate again).
* after unlock, weblate may add new commits (if the translators make some changes).